dts/arm/st/g4/stm32g431.dtsi

```diff
@@ -4,4 +4,4 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <st/g4/stm32g4.dtsi>
+#include <st/g4/stm32g43X.dtsi>
```

Comment:
I'm not fond of adding this intermediary files (stm32g43X and stm32g47X).
Also, I'd prefer we populate g4.dtsi with one version of the nodes and we overwrite the diverging property when required (say in stm32g474.dtsi), similarly to https://github.com/zephyrproject-rtos/zephyr/pull/30784.